src/App.jsx

```diff
@@ -14,45 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [currentSort, setSort] = useState(null);
+  const [revers, setReverse] = useState(null);
+
+  const SORT_ABC = 'abc';
+  const SORT_LENGTH = 'length';
+  const SORT_RESET = 'reset';
+
+  const sortBy = (sortType) => {
+    setSort(sortType);
+
+    switch (sortType) {
+      case SORT_ABC:
+        if (revers) {
+          setGoods([...goods].sort().reverse());
```

Comment:
you also can create a variable - a copy of the array. do not repeat the spread every time